workers/python/kitchen_sink.py

```diff
@@ -168,10 +214,44 @@ def launch_activity(execute_activity: ExecuteActivityAction) -> Awaitable:
                 execute_activity.remote.cancellation_type
             ),
         )
-    # TODO: Handle cancels
+
     return activity_task
 
 
+async def handle_awaitable_choice(
+    awaitable: Union[Coroutine | asyncio.Task], choice: AwaitableChoice
+):
+    if isinstance(awaitable, asyncio.Task):
+        task = awaitable
+    else:
+        # Place the awaitable into a task so we can cancel it easily
+        task = asyncio.create_task(awaitable)
+
+    did_cancel = False
+    try:
+        if choice.HasField("abandon"):
+            # Do nothing
+            return
+        elif choice.HasField("cancel_before_started"):
+            task.cancel()
+            did_cancel = True
+            await task
+        elif choice.HasField("cancel_after_started"):
+            await asyncio.sleep(0.001)
+            task.cancel()
+            did_cancel = True
+            await task
+        elif choice.HasField("cancel_after_completed"):
+            await task
+            task.cancel()
+            did_cancel = True
+        else:
+            await task
+    except asyncio.CancelledError:
+        if not did_cancel:
+            raise exceptions.ApplicationError("Unexpected cancellation")
```

Comment:
```suggestion
            raise
```

Might as well just re-raise the same thing like Go